### PR TITLE
docs(spin): Adds canary cmd docs.

### PR DESCRIPTION
**Pull request by jtk54**
@louisjimenez @dorbin PTAL.

### CODE REVIEW

**Review by louisjimenez — APPROVED** [review]
:+1:

File: guides/spin/canary-configs/index.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+title:  "Manage Canary Configs"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+## Overview
+
+`spin` can manage the whole lifecycle of your canary configs:
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
You can use `spin` to manage the whole lifecycle of your canary configs:
```

File: guides/spin/canary-configs/index.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      --gate-endpoint string     Gate (API server) endpoint (default http://localhost:8084)
+  -k, --insecure                 ignore certificate errors
+      --no-color                 disable color (default true)
+      --output string            configure output formatting
+  -q, --quiet                    squelch non-essential output
+
+Use " canary canary-config [command] --help" for more information about a command.
+```
+
+The following assumes Spinnaker is running and Gate is listening on `http://localhost:8084`. If
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
The following instructions assume Spinnaker is running and Gate is listening on `http://localhost:8084`. If
```

File: guides/spin/canary-configs/index.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+gate is running elsewhere, you can set the Gate endpoint with the global `--gate-endpoint` flag.
+
+## Managing Your Canary Config's Lifecycle
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion

```

File: guides/spin/canary-configs/index.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+### Create and update canary configs with `save`
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
## Create and update canary configs with `save`
```

File: guides/spin/canary-configs/index.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+```bash
+$ spin canary canary-configs save --file <path to canary config json>
+```
+
+Note that `save` accepts the canary config in JSON format. You can quickly export an existing
+canary config into a valid argument to the `--file` flag by using the `get` command.
+
+### List canary configs with `list`
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
## List canary configs with `list`
```

File: guides/spin/canary-configs/index.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  "id": "canaryConfigId",
+  "name": "canaryConfigName"
+  ...
+},
+...
+]
+
+```
+
+### Retrieve a single canary config with `get`
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
## Retrieve a single canary config with `get`
```

File: guides/spin/canary-configs/index.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+```bash
+spin canary canary-config get canaryConfigId
+{
+  "id": "myPipelineTemplate"
+  ...
+}
+```
+
+### Delete a canary config with `delete`
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
## Delete a canary config with `delete`
```

File: guides/spin/canary-configs/index.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+```bash
+spin canary canary-config delete canaryConfigId
+```
+
+### Test a canary config with `retro`
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md]:
```suggestion
## Test a canary config with `retro`
```

File: guides/spin/canary-configs/index.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+## Managing Your Canary Config's Lifecycle
+
+### Create and update canary configs with `save`
+
+```bash
+$ spin canary canary-configs save --file <path to canary config json>
+```
+
+Note that `save` accepts the canary config in JSON format. You can quickly export an existing
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md:54]: IIUC...

Would it work to say
`$ spin canary canary-configs save|get --file <path to canary config json>`

And then the explanation would be
Where:
* `save` accepts the canary config in JSON format from the file identified using the `--file` argument
* `get` exports an existing canary config into the identified file.

File: guides/spin/canary-configs/index.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+},
+...
+]
+
+```
+
+### Retrieve a single canary config with `get`
+
+```bash
+spin canary canary-config get canaryConfigId
```
Comment (dorbin) [inline guides/spin/canary-configs/index.md:77]: * The `save` version above is `canary-configs` (plural). This one is singular
* Is this different from the `get` version mentioned in that same place? That one takes a `--file`, but this one takes `canaryConfigId`, but are we doing the same thing here?

**Review by dorbin — COMMENTED** [review]
A few comments, questions, suggestions. 
After this is merged, I might edit that first section so it's doesn't look like cmd line help.

File: guides/spin/canary-configs/index.md (hunk 9 above)
Comment (jtk54) [reply guides/spin/canary-configs/index.md:54]: Sure

File: guides/spin/canary-configs/index.md (hunk 10 above)
Comment (jtk54) [reply guides/spin/canary-configs/index.md:77]: > That one takes a --file

It's the same `get. The `--file` belongs to `save`. That sentence was explaining how to use `get` as an argument to the `--file` param of `save.

**Comment by jtk54** [comment]
Which branch do the suggestion commits go to? I accepted them and have no idea how to rebase my local branch to include them.

File: guides/spin/canary-configs/index.md (hunk 9 above)
Comment (jtk54) [reply guides/spin/canary-configs/index.md:54]: Care to make these in a follow-up? We could also make similar changes in `pipeline-templates`, etc.

Comment (dorbin) [reply guides/spin/canary-configs/index.md:54]: Yeah, I can do that.

**Comment by dorbin** [comment]
@jtk54 I'm not sure. If I've been on the receiving end of this (new) suggestions feature, I don't remember how I got them incorporated.

File: guides/spin/canary-configs/index.md (hunk 9 above)
Comment (jtk54) [reply guides/spin/canary-configs/index.md:54]: Just to clarify also:

> get exports an existing canary config into the identified file.

`get` just dumps the json to stdout, but you can save that in a file if you want. There's no explicit output file for `get`.

**Comment by jtk54** [comment]
I sorted it out, I just had to pull my jtk54/whatever-branch branch. I think there is a delay between accepting the suggestions and it being synced back to my repo. In any case, we're good.

**Pull request merged**